AllFlyoutsBrokenOnAlarmsDetailsPage

swetha571 opened the pull request:
# Type of change? <!-- [x] all the boxes that apply -->

- [x] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
Fixed all flyouts in alarm Details Page

...

**Checklist:**

- [ ] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

spryor commented on src/components/maintenance/ruleDetails.js:
```diff
@@ -186,12 +186,7 @@ class RuleDetailsPage extends Component {
       rowData: this.state.devicesGridData,
       domLayout: 'autoHeight',
-      softSelectId: this.state.softSelectedDeviceId,
-      getSoftSelectId: btnActions.getSoftSelectId,
```
If you remove this, the soft selection state will not work. Why does this need to be removed? Are you sure you need to remove all of this?

swetha571 replied:
Yes. I am very sure. From morning investigated various ways. Later found the event handlers are causing the issue.

timlaverty replied:
Does it break soft selection?  i.e. we can't have hard select make the schedule/tags/properties dialogs work and in the process break the device details flyout.

spryor replied:
It doesn't break soft selection as the flyout still on opens. It appears to break the soft select effect on the row (the highlight). As it turns out, the soft select effect on this page is already broken since it wasn't implemented properly.

swetha571 commented:
I fixed the softSelect. I removed this because "onHardSelectChange: btnActions.onDeviceGridHardSelectChange," this event is coming from devicesGrid basic  (
![image](https://user-images.githubusercontent.com/16091688/32208826-47384f0e-bdc2-11e7-8a13-275b2def08d1.png))functionality.
